StyleCop.Analyzers/StyleCop.Analyzers/NamingRules/SA1311CodeFixProvider.cs

```diff
@@ -43,8 +47,58 @@ public override ImmutableArray<string> GetFixableDiagnosticIds()
                     continue;
 
                 var newName = char.ToUpper(token.ValueText[0]) + token.ValueText.Substring(1);
-                context.RegisterFix(CodeAction.Create($"Rename field to '{newName}'", cancellationToken => RenameHelper.RenameSymbolAsync(document, root, token, newName, cancellationToken)), diagnostic);
+
+                var typeSyntax = this.GetParentTypeDeclaration(token);
+
+                if (typeSyntax != null)
+                {
+                    SemanticModel semanticModel = await document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
+                    var typeDeclarationSymbol = semanticModel.GetDeclaredSymbol(typeSyntax) as INamedTypeSymbol;
+                    if (typeDeclarationSymbol == null)
+                    {
+                        return;
+                    }
+
+                    while (!this.IsValidNewMemberName(typeDeclarationSymbol, newName) || newName == typeDeclarationSymbol.Name)
```

Comment:
It's OK for it to propose a name that conflicts. It just means the user is responsible for choosing an appropriate name.

The user experience surrounding conflicting names is unpleasant, but that's another issue that will likely need to wait for a change in Roslyn.